src/browser/atom-window.coffee

```diff
@@ -73,7 +73,8 @@ class AtomWindow
     @browserWindow.loadUrl @getUrl(loadSettings)
     @browserWindow.focusOnWebView() if @isSpec
 
-    @openLocations(locationsToOpen) unless @isSpecWindow()
+    hasPathToOpen = not (locationsToOpen.length is 1 and not locationsToOpen[0].pathToOpen?)
```

Comment:
Also, I want to minimize edits to the browser code. Changes to it are scary as it's more on the opaque and hard-to-test side of things.